File: snies-backend/wellbeing_activities/domain/ports/wellbeing_activity_repository.py
from abc import ABC, abstractmethod

from wellbeing_activities.domain.entities.wellbeing_activity import WellbeingActivity


class WellbeingActivityRepository(ABC):
    @abstractmethod
    def create(self, activity: WellbeingActivity) -> WellbeingActivity:
        pass

    @abstractmethod
    def list(self) -> list[WellbeingActivity]:
        pass

    @abstractmethod
    def get_by_id(self, id: int) -> WellbeingActivity | None:
        pass

    @abstractmethod
    def update(self, id: int, activity: WellbeingActivity) -> WellbeingActivity | None:
        pass

    @abstractmethod
    def delete(self, id: int) -> bool:
        pass

File: snies-frontend/modules/wellbeing/hooks/ports/wellbeing-activity.repository.ts
import { WellbeingActivity } from "../../types/activity";
import { CreateWellbeingActivityInput } from "../types/create-activity-input";
export interface WellbeingActivityRepository {
    create(input: CreateWellbeingActivityInput): Promise<WellbeingActivity>;
    list(): Promise<WellbeingActivity[]>;
}
